### DISCUSSION TITLE
Unable to display custom validation error message

### DISCUSSION THREAD

**Discussion by toxifiedmatrix** [livewire/livewire#2184] [1. Help]
I am successfully able validate email input from two separate tables, i.e. users and invitations. The rule checks here for the unique email address, from both the tables. As it validates, it throws the same error message, for both the conditions. I need to display separate custom error message for both the conditions.

```
public Invitation $invitation;

protected $rules = [
    'invitation.email' => ['required', 'string', 'email', 'max:255', 'unique:invitations,email', 'unique:users,email'],
];

protected $messages = [
    'invitation.email.unique.invitations' => 'The email has already been invited.',
    'invitation.email.unique.users' => 'An account with this email has already been registered.',
];
```

**Comment by Silver343** [comment]
What is the error message that both are displaying?

**Reply by toxifiedmatrix** [reply]
@Silver343 No error message. It's just displaying the default Laravel error message. The error only gets visible, only when `invitation.email.unique.invitations` is reframed to `invitation.email.unique`. But here as you can see, I am passing the unique rule twice, one refers to the email column in the users table and the other refers to the email column in invitations table. So accordingly I need to display separate error messages for separate unique rule conditions.

**Comment by Senrism** [comment]
i think you can do it like this
protected $messages = [
    'invitation.email.invitations.unique' => 'The email has already been invited.',
    'invitation.email.users.unique' => 'An account with this email has already been registered.',
];

**Reply by daverydan** [reply]
I think you have to create **$rules** as well when using the **$messages** property.

Would you then use  it like so? 

**$this->validate($rules, $messages);**

**Reply by joshhanley** [reply]
@daverydan actually you can just use `$this->validate()` provided you have a `$messages` property on your component.

Check out "Customize Error Message & Attributes" section here in the docs https://laravel-livewire.com/docs/2.x/input-validation

Hope this helps!

**Comment by wcaaan** [comment]
$input = ['permission_name' => $this->resource_name . '.' . $this->permission_type];
					$rules = ['permission_name' => 'unique:permissions,name'];
					$messages = ['permission_name.unique' => 'The permission name ' . $this->resource_name . '.' . $this->permission_type . 'already exists.'];
					$validatedData = Validator::make($input, $rules, $messages)->validate();


why does this validation not working the error message say  "The given data was invalid"
